packages/webapp/src/views/DepartmentSummary/types.ts

```diff
@@ -0,0 +1,232 @@
+ * The selected focus area that should be used when the view first mounts. This is usually derived
+ * from the URL route.
+ */
+export type TstartingSelectedFocusArea = string;
+
+// Type: TfocusArea
+/**
+ * A single focus area associated with the consolidated budget.
+ */
+export type TfocusArea = string;
+
+// Type: TyearAndFocusAreaObject
+/**
+ * A set of props passed to `TchangeUrl` and `TgetValidYears`.
+ */
+export type TyearAndFocusAreaObject = {
+  year: TfinancialYear;
+  focusArea: TfocusArea;
+};
+
+// Type: TchangeUrl
+/**
+ * A callback that fires outside of the component each time a URL change of the should be triggered.
+ */
+export type TonUrlChange = (TyearAndFocusAreaObject) => void;
+
+// Type: TgetValidYears
+/**
+ * A function returned by `TconstructGetValidYearsFnProps` used to fetch the data required to
+ * calculate what years are valid for a specific focus area.
+ */
+export type TgetValidYears = (TyearAndFocusAreaObject) => void;
+
+// Type: TisValidYear
+/**
+ * Whether a specific focus area is present in a year or whether information for that year is
+ * available at all.
+ */
+export type TisValidYear = boolean;
+
+// Type: TisValidYearsList
+/**
+ * A list of `TisValidYear` items. Used enable toggling to a specific year in the year dropdown.
+ */
+export type TisValidYearsList = [TisValidYear, TisValidYear, TisValidYear, TisValidYear];
+
+// Type: TsetValidYearsState
+/**
+ * Sets the valid years state in the component.
+ */
+export type TsetValidYearsState = (TisValidYearsList) => void;
+
+// Type: TsetValidYearsState
+/**
+ * Sets the valid years state in the component.
+ */
+export type TsetErrorState = (Terror) => void;
+
+// Type: Tloading
+/**
+ * Whether a piece of data is still in the process of being fetched from the back-end API.
+ */
+export type Tloading = boolean;
+
+// Type: TsetLoading
+/**
+ * Sets the state for whether a specific piece of data is currently being fetched from the back-end API yet.
+ */
+export type TsetLoading = (Tloading) => void;
+
+// Type: TconstructGetValidYearsFnProps
+/**
+ * The props accepted by `TconstructGetValidYearsFn`.
+ */
+export type TconstructGetValidYearsFnProps = {
+  setValidYears: TsetValidYearsState;
+  setValidYearsLoading: TsetLoading;
+  setError: TsetErrorState;
+  latestYear: TlatestYear;
+};
+
+// Type: TconstructGetValidYearsFn
+/**
+ * A function that creates `TgetValidYears` through currying. Is passed from outside the component.
+ */
+export type TconstructGetValidYearsFn = (TconstructGetValidYearsFnProps) => TgetValidYears;
+
+// Type: TgetDepartmentsDataProps
+/**
+ * Props accepted by `TgetDepartmentsData`.
+ */
+export type TgetDepartmentsDataProps = {
+  year: TfinancialYear;
+};
+
+// Type: TgetDepartmentsData
+/**
+ * A function call that returns all the data required to render a focus area for a specific `TfinancialYear`.
+ */
+export type TgetDepartmentsData = (TgetDepartmentsDataProps) => void;
+
+// Type: TconstructGetFocusAreaDataProps
```

Comment:
Guys, I don't think these types.ts files is the way to go.

There's a reason why we tend to put types by the functions and methods where they're used. Code locality is a Big Thing. 

I'm jumping around about 5 files trying to figure out where the missing piece is and the types are in their own file?
Is this the goal, or is this just a stopgap to start adding types but eventually they'll be located adjacent to the relevant function?